File: binding/src/printsupport/libqpagesetupdialog.cpp
#include <QDialog>
#include <QObject>
#include <QPageSetupDialog>
#include <QPrinter>
#include <QString>
#include <QByteArray>
#include <cstring>
#include <QWidget>
#include <qpagesetupdialog.h>
#include "libqpagesetupdialog.h"
#include "libqpagesetupdialog.hxx"

QPageSetupDialog* QPageSetupDialog_new(QWidget* parent) {
	 return new VirtualQPageSetupDialog(parent);
}

QPageSetupDialog* QPageSetupDialog_new2(QPrinter* printer) {
	 return new VirtualQPageSetupDialog(printer);
}

QPageSetupDialog* QPageSetupDialog_new3() {
	 return new VirtualQPageSetupDialog();
}

QPageSetupDialog* QPageSetupDialog_new4(QPrinter* printer, QWidget* parent) {
	 return new VirtualQPageSetupDialog(printer, parent);
}

libqt_string QPageSetupDialog_Tr(const char* s) {
	QString _ret = QPageSetupDialog::tr(s);
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

void QPageSetupDialog_SetVisible(QPageSetupDialog* self, bool visible) {
	self->setVisible(visible);
}

int QPageSetupDialog_Exec(QPageSetupDialog* self) {
	return self->exec();
}

void QPageSetupDialog_Open(QPageSetupDialog* self, QObject* receiver, const char* member) {
	self->open(receiver, member);
}

void QPageSetupDialog_Done(QPageSetupDialog* self, int result) {
	self->done(result);
}

QPrinter* QPageSetupDialog_Printer(QPageSetupDialog* self) {
	return self->printer();
}

libqt_string QPageSetupDialog_Tr2(const char* s, const char* c) {
	QString _ret = QPageSetupDialog::tr(s, c);
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

libqt_string QPageSetupDialog_Tr3(const char* s, const char* c, int n) {
	QString _ret = QPageSetupDialog::tr(s, c, n);
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

// Base class handler implementation
void QPageSetupDialog_QBaseSetVisible(QPageSetupDialog* self, bool visible) {
	auto* vqpagesetupdialog = dynamic_cast<VirtualQPageSetupDialog*>(self);
	if (vqpagesetupdialog && vqpagesetupdialog->isVirtualQPageSetupDialog) {
vqpagesetupdialog->setQPageSetupDialog_SetVisible_IsBase(true);
	vqpagesetupdialog->setVisible(visible);
}
}

// Auxiliary method to allow providing re-implementation
void QPageSetupDialog_OnSetVisible(QPageSetupDialog* self, intptr_t slot) {
	auto* vqpagesetupdialog = dynamic_cast<VirtualQPageSetupDialog*>(self);
	if (vqpagesetupdialog && vqpagesetupdialog->isVirtualQPageSetupDialog) {
vqpagesetupdialog->setQPageSetupDialog_SetVisible_Callback(reinterpret_cast<VirtualQPageSetupDialog::QPageSetupDialog_SetVisible_Callback>(slot));
}
}

// Base class handler implementation
int QPageSetupDialog_QBaseExec(QPageSetupDialog* self) {
	auto* vqpagesetupdialog = dynamic_cast<VirtualQPageSetupDialog*>(self);
	if (vqpagesetupdialog && vqpagesetupdialog->isVirtualQPageSetupDialog) {
vqpagesetupdialog->setQPageSetupDialog_Exec_IsBase(true);
	return vqpagesetupdialog->exec();
}
}

// Auxiliary method to allow providing re-implementation
void QPageSetupDialog_OnExec(QPageSetupDialog* self, intptr_t slot) {
	auto* vqpagesetupdialog = dynamic_cast<VirtualQPageSetupDialog*>(self);
	if (vqpagesetupdialog && vqpagesetupdialog->isVirtualQPageSetupDialog) {
vqpagesetupdialog->setQPageSetupDialog_Exec_Callback(reinterpret_cast<VirtualQPageSetupDialog::QPageSetupDialog_Exec_Callback>(slot));
}
}

// Base class handler implementation
void QPageSetupDialog_QBaseDone(QPageSetupDialog* self, int result) {
	auto* vqpagesetupdialog = dynamic_cast<VirtualQPageSetupDialog*>(self);
	if (vqpagesetupdialog && vqpagesetupdialog->isVirtualQPageSetupDialog) {
vqpagesetupdialog->setQPageSetupDialog_Done_IsBase(true);
	vqpagesetupdialog->done(result);
}
}

// Auxiliary method to allow providing re-implementation
void QPageSetupDialog_OnDone(QPageSetupDialog* self, intptr_t slot) {
	auto* vqpagesetupdialog = dynamic_cast<VirtualQPageSetupDialog*>(self);
	if (vqpagesetupdialog && vqpagesetupdialog->isVirtualQPageSetupDialog) {
vqpagesetupdialog->setQPageSetupDialog_Done_Callback(reinterpret_cast<VirtualQPageSetupDialog::QPageSetupDialog_Done_Callback>(slot));
}
}

void QPageSetupDialog_Delete(QPageSetupDialog* self) {
    delete self;
}

File: binding/src/printsupport/libqpagesetupdialog.zig
const QtC = @import("qt6zig");
const qtc = @import("qt6c");%%_IMPORTLIBS_%%
%%_STRUCTDEFS_%%

/// https://doc.qt.io/qt-6/qpagesetupdialog.html
pub const qpagesetupdialog = struct {

    /// New constructs a new QPageSetupDialog object.
    pub fn New(parent: ?*anyopaque) QtC.QPageSetupDialog {
        return qtc.QPageSetupDialog_new(@ptrCast(parent));
    }


    /// New2 constructs a new QPageSetupDialog object.
    pub fn New2(printer: ?*anyopaque) QtC.QPageSetupDialog {
        return qtc.QPageSetupDialog_new2(@ptrCast(printer));
    }


    /// New3 constructs a new QPageSetupDialog object.
    pub fn New3() QtC.QPageSetupDialog {
        return qtc.QPageSetupDialog_new3();
    }


    /// New4 constructs a new QPageSetupDialog object.
    pub fn New4(printer: ?*anyopaque, parent: ?*anyopaque) QtC.QPageSetupDialog {
        return qtc.QPageSetupDialog_new4(@ptrCast(printer), @ptrCast(parent));
    }


    pub fn Tr(s: []const u8, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QPageSetupDialog_Tr(@ptrCast(s));
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qpagesetupdialog.Tr: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    pub fn SetVisible(self: ?*anyopaque, visible: bool) void {
        qtc.QPageSetupDialog_SetVisible(@ptrCast(self), visible);
    }

    /// Allows for overriding the related default method
    pub fn OnSetVisible(self: ?*anyopaque, callback: *const fn (?*anyopaque, bool) callconv(.c) void) void {
        qtc.QPageSetupDialog_OnSetVisible(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    /// Base class method implementation
    pub fn QBaseSetVisible(self: ?*anyopaque, visible: bool) void {
        qtc.QPageSetupDialog_QBaseSetVisible(@ptrCast(self), visible);
    }

    pub fn Exec(self: ?*anyopaque, ) i32 {
        return qtc.QPageSetupDialog_Exec(@ptrCast(self));
    }

    /// Allows for overriding the related default method
    pub fn OnExec(self: ?*anyopaque, callback: *const fn (?*anyopaque) callconv(.c) i32) void {
        qtc.QPageSetupDialog_OnExec(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    /// Base class method implementation
    pub fn QBaseExec(self: ?*anyopaque, ) i32 {
        return qtc.QPageSetupDialog_QBaseExec(@ptrCast(self));
    }

    pub fn Open(self: ?*anyopaque, receiver: ?*anyopaque, member: []const u8) void {
        qtc.QPageSetupDialog_Open(@ptrCast(self), @ptrCast(receiver), @ptrCast(member));
    }

    pub fn Done(self: ?*anyopaque, result: i32) void {
        qtc.QPageSetupDialog_Done(@ptrCast(self), result);
    }

    /// Allows for overriding the related default method
    pub fn OnDone(self: ?*anyopaque, callback: *const fn (?*anyopaque, i32) callconv(.c) void) void {
        qtc.QPageSetupDialog_OnDone(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    /// Base class method implementation
    pub fn QBaseDone(self: ?*anyopaque, result: i32) void {
        qtc.QPageSetupDialog_QBaseDone(@ptrCast(self), result);
    }

    pub fn Printer(self: ?*anyopaque, ) QtC.QPrinter {
        return qtc.QPageSetupDialog_Printer(@ptrCast(self));
    }

    pub fn Tr2(s: []const u8, c: []const u8, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QPageSetupDialog_Tr2(@ptrCast(s), @ptrCast(c));
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qpagesetupdialog.Tr2: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    pub fn Tr3(s: []const u8, c: []const u8, n: i32, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QPageSetupDialog_Tr3(@ptrCast(s), @ptrCast(c), n);
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qpagesetupdialog.Tr3: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    /// Delete this object from C++ memory.
    pub fn QDelete(self: ?*anyopaque) void {
        qtc.QPageSetupDialog_Delete(@ptrCast(self));
    }
};
